Source/WebCore/platform/graphics/FontCascade.h

```diff
@@ -247,6 +247,18 @@ class FontCascade final : public CanMakeWeakPtr<FontCascade>, public CanMakeChec
 private:
     enum class ForTextEmphasisOrNot : bool { NotForTextEmphasis, ForTextEmphasis };
 
+    CodePath codePathPrivate(const TextRun& run, std::optional<unsigned> from = std::nullopt, std::optional<unsigned> to = std::nullopt) const
```

Comment:
I would use a more descriptive name, e.g. determineCodePath, and then rename codePath to determineCodePathForTextRun.

Reply:
The variable 'run' is used below. Can't leave it out.

Reply:
Oops sure you need run 🤣

Reply:
The idea of this change is stating different code paths for the external and internal.
FontCascade::codePath() is for the external, and returning a code path for the external code.
FontCascade::codePathPrivate() is for the internal, and returning a code path for the internal code.
The method name determineCodePathForTextRun() doesn't sound good to me.

Reply:
However, FontCascade has two `codePath` methods, `codePath()` and `codePath(const TextRun&, std::optional<unsigned>, std::optional<unsigned>) const`.
I think it's a good idea to rename the latter to `codePathForTextRun(const TextRun&, std::optional<unsigned>, std::optional<unsigned>) const`.
But, this is another topic. I don't want to do it in the patch.
IMHO, those `codePath` methods should be TextRun's method. And, TextRun should cache the result. But, I don't understand enough. I don't suggest that.

Reply:
You might not be happy with the naming proposal, that is ok:

```
FontCascade::codePath() is for the external, and returning a code path for the external code.
FontCascade::codePathPrivate() is for the internal, and returning a code path for the internal code.
```

But sorry, this is not clear from the naming of the functions. codePath() and codePathPrivate() is non-obvious, and it's not clear to anyone when which code path should be used. Maybe @carlosgcampos has a better suggestion?

Reply:
I still don't see the problem of codePathForShaping(), but I agree Private or Internal are confusing.

Reply:
Fixed.